optimum/intel/neural_compressor/modeling_base_seq2seq.py

```diff
@@ -0,0 +1,419 @@
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        encoder_file_name: Optional[str] = ENCODER_NAME,
+        decoder_file_name: Optional[str] = DECODER_NAME,
+        decoder_with_past_file_name: Optional[str] = DECODER_WITH_PAST_NAME,
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        torch_dtype: Optional[Union[str, "torch.dtype"]] = None,
+        **kwargs,
+    ):
+        """
+        Loads a model and its configuration file from a directory or the HF Hub.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+                Can be either:
+                    - The model id of a pretrained model hosted inside a model repo on huggingface.co.
+                    - The path to a directory containing the model weights.
+            use_auth_token (`str` or `bool`):
+                The token to use as HTTP bearer authorization for remote files. Needed to load models from a private
+                repository.
+            revision (`str`):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id.
+            force_download (`bool`, *optional*, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            cache_dir (`Union[str, Path]`, *optional*):
+                The path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            encoder_file_name(`str`, *optional*):
+                The encoder model file name. Overwrites the default file name and allows one to
+                load the encoder model with a different name.
+            decoder_file_name(`str`, *optional*):
+                The decoder model file name. Overwrites the default file name and allows one to
+                load the decoder model with a different name.
+            decoder_with_past_file_name(`str`, *optional*):
+                The decoder with past key values model file name overwriting the default file name,
+                allowing to load the decoder model with a different name.
+            local_files_only(`bool`, *optional*, defaults to `False`):
+                Whether or not to only look at local files (i.e., do not try to download the model).
+        """
+        model_save_dir = None
+        model_dim = None
+        model_parallel = None
+
+        decoder_with_past = None
+        if config.torchscript:
+            # Load model from a local directory
+            if os.path.isdir(model_id):
+                encoder = cls.load_model(os.path.join(model_id, encoder_file_name))
+                decoder = cls.load_model(os.path.join(model_id, decoder_file_name))
+                decoder_with_past = (
```

Comment:
Why do we need a `decoder_with_past` ? We should keep only one decoder which would handle both cache and not